.github/workflows/retest.yaml

```diff
@@ -0,0 +1,26 @@
+name: Detect and Trigger Retest
+on:
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  retest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        if: github.event.comment.author_association == 'MEMBER' && contains(github.event.comment.body, '/retest')
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b #v4.1.4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Retest
+        if: github.event.comment.author_association == 'MEMBER' && contains(github.event.comment.body, '/retest')
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: ${{ github.repository }}
+          COMMENT: ${{ github.event.comment.body }}
+        run: |
+          gh pr checkout ${{ github.event.issue.number }}
+          SHA_REF=$(git rev-parse HEAD)
```

Comment:
can we avoid doing the initial checkout and the branch change entirely? i.e. don't use `git`

I see there's an API method to get commits per PR: https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request

Reply:
Nice catch 🚀